gensim/similarities/docsim.py

```diff
@@ -392,12 +394,18 @@ def __iter__(self):
         # turn off query normalization (vectors in the index are already normalized, save some CPU)
         norm, self.norm = self.norm, False
 
-        for chunk in self.iter_chunks():
-            if chunk.shape[0] > 1:
-                for sim in self[chunk]:
+        with concurrent.futures.ProcessPoolExecutor() as executor:
+            futures = []
+            process_chunk = functools.partial(_query_chunk, index=self)
+
+            chunks = list(self.iter_chunks())
```

Comment:
Better avoid loading entire index into memory by loading the chunk inside the executor.